src/microbe_stage/MicrobeHUD.cs

```diff
@@ -838,4 +838,38 @@ private void OnTranslationsChanged()
         UpdateColonySizeForMulticellular();
         UpdateColonySizeForMacroscopic();
     }
+
+    private void ToggleProcessPressed(ChemicalEquation equation)
+    {
+        if (!stage!.Player.Has<BioProcesses>())
+            return;
+
+        ref var processes = ref stage.Player.Get<BioProcesses>();
+
+        if (processes.ActiveProcesses == null)
+            return;
+
+        var newProcesses = new List<TweakedProcess>();
+
+        foreach (var process in processes.ActiveProcesses)
```

Comment:
This logic seems a bit weird to me. Why doesn't this just loop the active processes and modify the speed multiplier of the process that matches the equation based on the `BioProcess` it does?

Reply:
Foreach elements cannot be modified while running the loop.

Reply:
I had some trouble with this one

Reply:
Use a `for (int i = 0; i < ...) ` loop

Reply:
"Cannot modify the return value of 'expression' because it is not a variable" `processes.ActiveProcesses[i].SpeedMultiplier = equation.ProcessEnabled ? 1 : 0;`

Reply:
I think you need to read the struct to a local variable, modify it and then write the full thing back to the list.

Reply:
![image](https://github.com/user-attachments/assets/32a15bda-cb9d-4d5b-b854-04ebaae03bfc)
This one doesn't work either

Reply:
You need:
```c#
var list = ....;
var count = list.Count;
for (int i = 0; i < count; ++i)
{
     if (list[i].Stuff == condition)
     {
            var process = list[i];
            process.Thing = updated thing;
            list[i] = process;
     }
}
```